### PR TITLE
Feature: Preserving Folder Structure 

### PR DESCRIPTION
**Pull request by r888800009**
* Refactor source/creator/io/package.d confusing code
* Add the KRA/PSD import "Preserve Folder Structure" configuration option
* Add DialogHandler

image

image

### CODE REVIEW

**Comment by LunaTheFoxgirl** [comment]
Right now this depends on a specific commit of kra-d; please make a PR to kra-d first and let me release a new version if you've made any fixes.

**Comment by r888800009** [comment]
> Right now this depends on a specific commit of kra-d; please make a PR to kra-d first and let me release a new version if you've made any fixes.

@LunaTheFoxgirl This fix is ​​based on the latest kra-d commit. You may need to run `dub upgrade` before build.

https://github.com/Inochi2D/kra-d/commit/6a0bd415c2bb948e4414e9e3b9d2113f1bf21e42
https://github.com/Inochi2D/kra-d/pull/1

**Comment by LunaTheFoxgirl** [comment]
> > Right now this depends on a specific commit of kra-d; please make a PR to kra-d first and let me release a new version if you've made any fixes.
> 
> @LunaTheFoxgirl This fix is ​​based on the latest kra-d commit. You may need to run `dub upgrade` before build.
> 
> [Inochi2D/kra-d@6a0bd41](https://github.com/Inochi2D/kra-d/commit/6a0bd415c2bb948e4414e9e3b9d2113f1bf21e42) [Inochi2D/kra-d#1](https://github.com/Inochi2D/kra-d/pull/1)

Ah, I missed that, please change the version specification to ~>0.5.6; that includes that change.

File: dub.sdl [hunk 1]
```diff
@@ -4,8 +4,8 @@ authors "Inochi2D Project"
 copyright "Copyright © 2020, Inochi2D Project"
 license "BSD 2-clause"
 dependency "psd-d" version="~>0.6.1"
-dependency "inmath" version="~>1.0.5"
-dependency "kra-d" version="~>0.5.5"
+dependency "inmath" version="1.0.6"
```
Comment (LunaTheFoxgirl) [inline dub.sdl]: this should be ~>1.0.6; to allow chosing other 1.0.x series if updates are released.

Alternatively, change to ==1.0.6 to make it limited to that specific version.

**Review by LunaTheFoxgirl — CHANGES_REQUESTED** [review]
Mostly looks good, just one problem left in dub.sdl

File: dub.sdl (hunk 1 above)
Comment (LunaTheFoxgirl) [reply dub.sdl]: Please see: https://dub.pm/dub-reference/build_settings/#version-specifiers

**Comment by LunaTheFoxgirl** [comment]
LGTM

**Review by LunaTheFoxgirl — APPROVED** [review]

**Pull request merged**